### PR TITLE
feat(theme): ng-add schematic

### PR DESCRIPTION
**Pull request by Tibing**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/nebular/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:

- ng-add schematics was added
- documentation was updated
- the build flow was slightly modified, so, please, check it

### CODE REVIEW

**Comment by Tibing** [comment]
@yggg, could you please check again updated [installation guide](https://github.com/akveo/nebular/blob/b7530f344a58522d762a35583fb6943807f56db8/docs/articles/install-into-existing.md)

File: src/framework/theme/schematics/ng-add/register-modules/index.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    const noopAnimationsModuleName = 'NoopAnimationsModule';
+    const animationsPackage = '@angular/platform-browser/animations';
+
+    if (options.animations) {
+      // In case the project explicitly uses the NoopAnimationsModule, we should print a warning
+      // message that makes the user aware of the fact that we won't automatically set up
+      // animations. If we would add the BrowserAnimationsModule while the NoopAnimationsModule
+      // is already configured, we would cause unexpected behavior and runtime exceptions.
+      if (hasNgModuleImport(tree, appModulePath, noopAnimationsModuleName)) {
+        return console.warn(red(`Could not set up "${bold(browserAnimationsModuleName)}" ` +
```
Comment (yggg) [inline src/framework/theme/schematics/ng-add/register-modules/index.ts]: context

File: src/framework/theme/schematics/ng-add/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * Add required peer dependencies in package.json if needed.
+ * */
+function registerPeerDependencies(host: Tree) {
+  const angularCoreVersion = getDependencyVersionFromPackageJson(host, '@angular/core');
+  const nebularThemeVersion = getNebularVersion();
+  const nebularIconsVersion = getNebularPeerDependencyVersionFromPackageJson('nebular-icons');
+
+  addPackageToPackageJson(host, '@angular/cdk', angularCoreVersion);
+  addPackageToPackageJson(host, '@angular/animations', angularCoreVersion);
+  addPackageToPackageJson(host, '@schematics/angular', angularCoreVersion);
```
Comment (yggg) [inline src/framework/theme/schematics/ng-add/index.ts]: Isn't `@schematics/angular` a dev dependency?

File: src/framework/theme/schematics/ng-add/register-modules/index.ts (hunk 1 above)
Comment (yggg) [inline src/framework/theme/schematics/ng-add/register-modules/index.ts]: context

File: src/framework/theme/schematics/ng-add/index.ts (hunk 2 above)
Comment (yggg) [inline src/framework/theme/schematics/ng-add/index.ts]: Isn't `@schematics/angular` a dev dependency?

**Pull request merged**